### PR TITLE
CTCTRADERS-remove-seals

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

File: app/controllers/events/seals/ConfirmRemoveSealController.scala [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                  val json = Json.obj(
+                    "form"       -> formWithErrors,
+                    "mode"       -> mode,
+                    "mrn"        -> mrn,
+                    "sealNumber" -> sealNumber,
+                    "radios"     -> Radios.yesNo(formWithErrors("value"))
+                  )
+
+                  renderer.render("events/seals/confirmRemoveSeal.njk", json).map(BadRequest(_))
+                case _ => Future.successful(Redirect(controllers.routes.SessionExpiredController.onPageLoad()))
```
Comment (amaalali) [inline app/controllers/events/seals/ConfirmRemoveSealController.scala]: Should we treat this as fine and show them a page saying something along the lines of _This has already been removed_ with a link to the next sensible page (ask design for what we should do)?

File: conf/app.routes [hunk 2]
```diff
@@ -152,10 +152,10 @@ POST       /:mrn/event/:eventIndex/haveSealsChanged
...
 
-GET        /:mrn/removeSeal                                             controllers.events.seals.RemoveSealController.onPageLoad(mrn: MovementReferenceNumber, mode: Mode = NormalMode)
-POST       /:mrn/removeSeal                                             controllers.events.seals.RemoveSealController.onSubmit(mrn: MovementReferenceNumber, mode: Mode = NormalMode)
-GET        /:mrn/changeRemoveSeal                                       controllers.events.seals.RemoveSealController.onPageLoad(mrn: MovementReferenceNumber, mode: Mode = CheckMode)
-POST       /:mrn/changeRemoveSeal                                       controllers.events.seals.RemoveSealController.onSubmit(mrn: MovementReferenceNumber, mode: Mode = CheckMode)
+GET        /:mrn/event/:eventIndex/seal/:sealIndex/removeSeal                                      controllers.events.seals.ConfirmRemoveSealController.onPageLoad(mrn: MovementReferenceNumber, eventIndex: Int, sealIndex: Int, mode: Mode = NormalMode)
```
Comment (chris-ianson) [inline conf/app.routes]: Can we have confirmRemoveSeal page here, please? Keeps uris consistent

File: app/controllers/events/seals/ConfirmRemoveSealController.scala (hunk 1 above)
Comment (chris-ianson) [reply app/controllers/events/seals/ConfirmRemoveSealController.scala]: Thanks Amaal. Did you mean the onPageLoad method (more likely to happen there) I've had a chat with design on this and they agree it seems logical to have a more informative error screen. Raises the question, we do it here where else should we do it (on remove containers and events also?) We're going to have a session later today to go over the spike which will feed into similar issues. We'll let you and Matt know how we get on when you're back in and get your thoughts

Comment (amaalali) [reply app/controllers/events/seals/ConfirmRemoveSealController.scala]: On the `GET` call (`onPageLoad`), that's a request to confirm to delete, so as you say that makes sense to still have an error that is informative. But on the `POST` they are requesting to delete something at an index that does not exist (they probably do this mostly through URL manipulation), so there no need to delete and we can tell them that and then link them off to the Add Seal page? I'm thinking of dual users here, so we don't tell them the data has been removed when it hasn't

**Pull request merged**